### ISSUE TITLE
[Bug]: DataGrid: Accessibility Insights: ARIA attribute is not allowed: aria-checked="false"

### ISSUE THREAD

**Issue by havinash123** [microsoft/fluentui#31073]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
When multiselect is enabled in a DataGrid component, it results in an accessibility issue:
Ensures ARIA attributes are allowed for an element's role
```

### Are you reporting Accessibility issue?

None

### Reproduction

It's even reproducible in the default example given in the official documentation:https://react.fluentui.dev/?path=/docs/components-datagrid--default

## Actual Behavior
Accessibility Error
## Expected Behavior
No Accessibility Error

![image](https://github.com/microsoft/fluentui/assets/167244397/fb84ec06-46de-48a7-a778-20a61171019f)

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by TristanWatanabe** [comment]
According to the latest[ aria spec](https://www.w3.org/TR/wai-aria/#aria-checked) - the `aria-checked` attribute which is being flagged by a11y insights on an element with `role=gridcell` is not valid and probably shouldn't be applied

**Comment by CharlesYWL** [comment]
Any updates or workaround on this issue?

**Comment by smhigley** [comment]
This is actually intentional; the selection attribute that is allowed on gridcells (`aria-selected`) does not work well for multiselect, and does not support mixed state.

While it's true that this is technically not allowed on gridcells (for now, we've actually discussed changing this in ARIA), it does no harm and does have some practical support, which is why we've left it on.
